agent-install.sh: zod not installed into shared ~/.openclaw/node_modules — agent_chat fails to load at runtime

## Summary

During SE run #94 staging test of `fix/installer-bugs-266-315-316`, the `agent-install.sh` installer fails at the agent_chat TypeScript build step with `exit 1`, preventing the installer from completing.

## Error Output

```
Building agent_chat TypeScript...
❌ Build failed
src/channel.ts(378,9): error TS2304: Cannot find name 'clearInterval'.
src/channel.ts(476,22): error TS7006: Parameter 'cfg' implicitly has an 'any' type.
src/channel.ts(487,22): error TS7006: Parameter 'cfg' implicitly has an 'any' type.
[... 18 more TS errors ...]
src/config.ts(1,19): error TS2307: Cannot find module 'zod' or its corresponding type declarations.
src/config.ts(2,42): error TS2307: Cannot find module 'openclaw/plugin-sdk' or its corresponding type declarations.
src/runtime.ts(1,36): error TS2307: Cannot find module 'openclaw/plugin-sdk' or its corresponding type declarations.
```

## Impact

Because `agent-install.sh` uses `set -euo pipefail` and the agent_chat build step does `exit 1` on failure (line ~1455), the installer exits before reaching:
- `install_metacognition_plugin` calls (self-awareness, confidence-check)
- Shell environment setup (NOVA_DIR, bash_env)

This prevents a clean end-to-end install in a fresh staging environment.

## Root Cause

Pre-existing issue — confirmed by:
1. Zero diff between `origin/main` and `origin/fix/installer-bugs-266-315-316` in `cognition/agent-chat/`
2. Errors indicate missing type declarations for `openclaw/plugin-sdk` and `zod` — the `node_modules` tree in a fresh clone does not have these available to the extension's tsconfig

## Environment
- Fresh clone of nova-mind on nova-staging
- OpenClaw 2026.6.6-nova (aa7ac2c)
- Node.js v22.22.3
- TypeScript via `npx tsc`

## Proposed Fix

Either:
1. Add `|| true` to the agent_chat TS build step so a build failure is non-fatal (logs warning, continues install), OR
2. Ensure `npm install` in the agent_chat extension properly resolves `openclaw/plugin-sdk` and `zod` dependencies before `tsc` runs

Discovered during: SE #94 / nova-mind staging test for fix/installer-bugs-266-315-316

--- a/agent-install.sh
+++ b/agent-install.sh
@@ -1482,10 +1482,24 @@ if [ -d "$EXTENSION_SOURCE" ]; then
         fi
     fi
 
+    # Install extension dependencies
+    echo ""
+    echo "  Installing agent_chat extension dependencies..."
+    cd "$EXTENSION_TARGET"
+
+    NPM_INSTALL_LOG="${TMPDIR:-/tmp}/npm-install-agent-chat-$$.log"
+    if npm install >"$NPM_INSTALL_LOG" 2>&1; then
+        echo -e "  ${CHECK_MARK} Extension dependencies installed"
+        rm -f "$NPM_INSTALL_LOG"
+    else
+        echo -e "  ${CROSS_MARK} npm install failed for agent_chat"
+        tail -20 "$NPM_INSTALL_LOG"
+        exit 1
+    fi
+
     # Build TypeScript
     echo ""
     echo "  Building agent_chat TypeScript..."
-    cd "$EXTENSION_TARGET"
 
     NPM_BUILD_LOG="${TMPDIR:-/tmp}/npm-build-agent-chat-$$.log"
     if npm run build >"$NPM_BUILD_LOG" 2>&1; then

--- a/cognition/focus/agent_chat/package.json
+++ b/cognition/focus/agent_chat/package.json
@@ -25,6 +25,7 @@
   "dependencies": {
     "@types/node": "^22.0.0",
     "@types/pg": "^8.11.0",
+    "openclaw": "2026.2.9",
     "pg": "^8.11.3",
     "typescript": "^5.6.0",
     "zod": "^3.23.0"

